### PR TITLE
Mypy types

### PR DESCRIPTION
**Pull request by olliethomas**

### CODE REVIEW

File: ethicml/evaluators/evaluate_models.py [hunk 1]
```diff
@@ -46,40 +46,37 @@ def evaluate_models(datasets: List[Dataset], preprocess_models: List[PreAlgorith
...
 
                 temp_res['model'] = model.name
 
-                if test_mode:
-                    predictions: np.array = model.run_test(train, test)
-                else:
-                    predictions: np.array = model.run(train, test)
+                predictions = model.run_test(transformed_train, transformed_test) \
+                    if test_mode \
+                    else model.run(transformed_train, transformed_test)
```
Comment (thomkeh) [inline ethicml/evaluators/evaluate_models.py]: yeah... so, I think

```python
                predictions: np.array
                if test_mode:
                    predictions = model.run_test(train_, test_)
                else:
                    predictions = model.run(train_, test_)
```
is a bit nicer to read. It's especially difficult to understand if the condition (`test_mode`) is on the line after the corresponding value.

**Comment by olliethomas** [comment]
I have no idea why this is failing

File: ethicml/data/compas.py [hunk 2]
```diff
@@ -473,10 +475,11 @@ def filename(self) -> str:
...
     def feature_split(self) -> Dict[str, List[str]]:
-        if self.discrete_only:
-            conc_features: List[str] = self.s_prefix + self.y_prefix + self.continuous_features
-        else:
-            conc_features: List[str] = self.s_prefix + self.y_prefix
+
+        conc_features: List[str] = self.s_prefix + self.y_prefix + self.continuous_features \
+            if self.discrete_only \
+            else self.s_prefix + self.y_prefix
+
```
Comment (thomkeh) [inline ethicml/data/compas.py]:
```python
conc_features: List[str]
if self.discrete_only:
    conc_features = self.s_prefix + self.y_prefix + self.continuous_features
else:
    conc_features = self.s_prefix + self.y_prefix
```

File: ethicml/data/adult.py [hunk 3]
```diff
@@ -224,10 +224,9 @@ def filename(self) -> str:
 
     @property
     def feature_split(self) -> Dict[str, List[str]]:
-        if self.discrete_only:
-            conc_features: List[str] = self.s_prefix+self.y_prefix+self.continuous_features
-        else:
-            conc_features: List[str] = self.s_prefix + self.y_prefix
+        conc_features: List[str] = self.s_prefix + self.y_prefix + self.continuous_features \
+            if self.discrete_only \
+            else self.s_prefix + self.y_prefix
```
Comment (thomkeh) [inline ethicml/data/adult.py]:
```python
conc_features: List[str]
if self.discrete_only:
    conc_features = self.s_prefix + self.y_prefix + self.continuous_features
else:
    conc_features = self.s_prefix + self.y_prefix
```

**Pull request merged**